ush/remap.sh

```diff
@@ -0,0 +1,376 @@
+#   The cleaned string with no whitespace.
+#
+# Example usage:
+#   cleaned_string=$(_strip_whitespace "   This is a string   with spaces  ")
+#   echo "Cleaned string: \"$cleaned_string\""
+#
+# This example will remove all leading, trailing, and internal
+# whitespace from the input string and display the cleaned result.
+function _strip_whitespace(){
+    local in_string="${1}"
+
+    out_string=$(echo "${in_string}" | sed "s/ //g")
+}
+
+#######
+
+# nc_concat - Concatenate netCDF files and manage file paths.
+#
+# Description:
+#   This function performs the concatenation of two netCDF files:
+#   `output_path` and `var_interp_path`. If the `output_path` file
+#   exists, the function merges `var_interp_path` into it and manages
+#   temporary files accordingly. If `output_path` does not exist, it
+#   creates a new netCDF file by renaming `var_interp_path` to
+#   `output_path`.
+#
+# Parameters:
+#   $1 - The path to the variable-interpolated netCDF file.
+#
+# Global Variables:
+#   output_path - The path to the output netCDF file.
+#
+# Example usage:
+#   nc_concat "variable_interpolated.nc"
+#
+# This example will concatenate the variable-interpolated file into
+# the output netCDF file, handling the case where the output file may
+# or may not exist.
+function nc_concat(){
+    local var_interp_path="${1}"
+
+    tmp_nc_file="${PWD}/tmp_nc.nc"
+    if test -e "${output_path}"; then
+	echo "netCDF-formatted file path ${output_path} exists; merging ${var_interp_path}"
+	cdo merge "${output_path}" "${var_interp_path}" "${tmp_nc_file}"
+	mv "${tmp_nc_file}" "${output_path}"
+	rm "${var_interp_path}" >> /dev/null
+    else
+	echo "netCDF-formatted file path ${output_path} does not exist; creating..."
+	mv "${var_interp_path}" "${output_path}"
+    fi
+}
+
+#######
+
+# cdo_remap - Perform CDO remapping and concatenate the result.
+#
+# Description:
+#   This function remaps a specific variable `varname` from an input
+#   netCDF file `varfile` using the specified interpolation method
+#   `interp_type`. It creates an intermediate netCDF file
+#   `varname.interp.nc` containing the remapped variable, and then it
+#   concatenates this intermediate file into the output netCDF file
+#   defined by the global variable `output_path`.
+#
+# Parameters:
+#   $1 - The name of the variable to remap.
+#   $2 - The path to the input netCDF file.
+#   $3 - The interpolation method to use (e.g., "remapbil").
+#
+# Global Variables:
+#   output_path - The path to the output netCDF file.
+#   dstgrid_config - The configuration for destination grid specifications.
+#
+# Example usage:
+#   cdo_remap "temperature" "input_file.nc" "remapbil"
+#
+# This example remaps the "temperature" variable using bilinear
+# interpolation from the input file and appends the result to the
+# output netCDF file.
+function cdo_remap(){
+    local varname="${1}"
+    local varfile="${2}"
+    local interp_type="${3}"
+    local var_interp_path="${PWD}/${varname}.interp.nc"
```

Comment:
`${PWD}` is undefined.

Reply:
This is defined in the environment once a shell is invoked.

Here are examples from three different hosts.

- RDHPCS Hera: 
   `[Henry.Winterbottom@hfe07 Neil.Barton]$ echo $PWD
     /scratch1/NCEPDEV/da/Henry.Winterbottom/scratch
    `
- NOAA CSP AWS:
    `[Henry.Winterbottom@globalworkflowtest-3 logs]$ echo $PWD
     /contrib/Rajendra.Panda/global-workflow/sorc/logs
    `
- OSX:
   `[leviathan:UFS/global-workflow/ush] henry.winterbottom% echo $PWD
    /Users/henry.winterbottom/trunk/UFS/global-workflow/ush
   `
- Ubuntu 20.04 Docker container:
   `root@a7c3ca831789:/home# echo $PWD
    /home
   `

Reply:
Okay. Somehow I was unaware of this shell shortcut, I've always used `$(pwd)`. But apparently `${PWD}` is faster.